Modules/Devices/src/Devices.Domain/Entities/Identities/Identity.cs

```diff
@@ -165,12 +172,31 @@ public void DeletionGracePeriodReminder3Sent()
     {
         return DeletionProcesses.FirstOrDefault(x => x.Status == deletionProcessStatus);
     }
+
+    public IdentityDeletionProcess CancelDeletionProcess(IdentityDeletionProcessId deletionProcessId, DeviceId canceledByDeviceId)
+    {
+        EnsureIdentityOwnsDevice(canceledByDeviceId);
+
+        var deletionProcess = DeletionProcesses.FirstOrDefault(x => x.Id == deletionProcessId) ??
+                              throw new DomainException(GenericDomainErrors.NotFound(nameof(IdentityDeletionProcess)));
+
+        if (Status == IdentityStatus.Active)
+            throw new DomainException(DomainErrors.DeletionProcessGracePeriodHasEnded((DateTime)deletionProcess.GracePeriodEndsAt));
```

Comment:
I think we can even remove this check completely and instead rely on the `deletionProcess.Cancel method` to throw.
Remember to add a corresponding test for this.